rti_connext_dds_cmake_module/cmake/rti_build_helper.cmake

```diff
@@ -529,18 +526,27 @@ function(rti_guess_connextdds_arch)
       if(CONNEXTDDS_ARCH)
         break()
       endif()
+
     endforeach()
-  endif()
 
-  if(NOT CONNEXTDDS_ARCH)
-    message(WARNING
-      "CONNEXTDDS_ARCH not specified. Please set "
-      "-DCONNEXTDDS_ARCH= to specify your RTI Connext DDS "
-      " architecture")
-  else()
+    if(NOT CONNEXTDDS_ARCH)
+      list(GET architecture_candidates 0 CONNEXTDDS_ARCH)
```

Comment:
what if the `architectures_installed` is empty, is it safe to get the item from `architecture_candidates`?

Reply:
in this case the CONNEXTDDS_ARCH will be not set so will print out the message in line 540, which means the user didn't install any non-foreign architectures that would fit their operation system. The build will fail in this case and the original logical both.

Reply:
I don't think the following error message is expected.

```
CMake Error at CMakeLists.txt:13 (list):
  list GET given empty list
```

Reply:
sorry for not examining carefully, i made the change for checking it